src/base_client.rs

```diff
@@ -49,6 +48,20 @@ pub struct BaseClient {
 
     // Latest account sequence
     pub account_sequence: Option<u64>,
+
+
+}
+
+#[derive(Debug)]
+pub enum GasConfig {
```

Comment:
Very *very* minor nit on wording: I'd suggest to use "Policy" instead of "Config" because somehow concept of "gas config" sounds weird to me.

For funs, you could even use "FuelPolicy" :stuck_out_tongue:  but I guess "GasPolicy" is most obvious?